invenio_collections/ext.py

```diff
@@ -38,10 +40,17 @@ class _AppState(object):
     def __init__(self, app, cache=None):
         """Initialize state."""
         self.app = app
-        self.cache = cache
+        self._cache = cache
         if self.app.config['COLLECTIONS_REGISTER_RECORD_SIGNALS']:
             self.register_signals()
 
+    @cached_property
+    def cache(self):
+        """Return a cache instance."""
+        cache = self._cache or self.app.config.get('COLLECTIONS_CACHE')
+        return import_string(cache) if instance(cache, six.string_types) \
+            else cache
```

Comment:
`None` should be fine, although it would be better to return [`NullCache`](http://werkzeug.pocoo.org/docs/0.11/contrib/cache/#werkzeug.contrib.cache.NullCache) in that case.